src/sse/sse_butterflies.rs

```diff
@@ -0,0 +1,4246 @@
+        let temp_a2_1 = _mm_mul_pd(self.twiddle2re, x14p);
+        let temp_a2_2 = _mm_mul_pd(self.twiddle1re, x23p);
+
+        let temp_b1_1 = _mm_mul_pd(self.twiddle1im, x14n);
+        let temp_b1_2 = _mm_mul_pd(self.twiddle2im, x23n);
+        let temp_b2_1 = _mm_mul_pd(self.twiddle2im, x14n);
+        let temp_b2_2 = _mm_mul_pd(self.twiddle1im, x23n);
+
+        let temp_a1 = _mm_add_pd(value0, _mm_add_pd(temp_a1_1, temp_a1_2));
+        let temp_a2 = _mm_add_pd(value0, _mm_add_pd(temp_a2_1, temp_a2_2));
+
+        let temp_b1 = _mm_add_pd(temp_b1_1, temp_b1_2);
+        let temp_b2 = _mm_sub_pd(temp_b2_1, temp_b2_2);
+
+        let temp_b1_rot = self.rotate.rotate(temp_b1);
+        let temp_b2_rot = self.rotate.rotate(temp_b2);
+
+        let x0 = _mm_add_pd(value0, _mm_add_pd(x14p, x23p));
+        let x1 = _mm_add_pd(temp_a1, temp_b1_rot);
+        let x2 = _mm_add_pd(temp_a2, temp_b2_rot);
+        let x3 = _mm_sub_pd(temp_a2, temp_b2_rot);
+        let x4 = _mm_sub_pd(temp_a1, temp_b1_rot);
+        [x0, x1, x2, x3, x4]
+    }
+}
+
+//    __             _________  _     _ _
+//   / /_           |___ /___ \| |__ (_) |_
+//  | '_ \   _____    |_ \ __) | '_ \| | __|
+//  | (_) | |_____|  ___) / __/| |_) | | |_
+//   \___/          |____/_____|_.__/|_|\__|
+//
+
+pub struct SseF32Butterfly6<T> {
+    direction: FftDirection,
+    _phantom: std::marker::PhantomData<T>,
+    bf3: SseF32Butterfly3<T>,
+}
+
+boilerplate_fft_sse_f32_butterfly!(SseF32Butterfly6, 6, |this: &SseF32Butterfly6<_>| this
+    .direction);
+boilerplate_fft_sse_common_butterfly!(SseF32Butterfly6, 6, |this: &SseF32Butterfly6<_>| this
+    .direction);
+impl<T: FftNum> SseF32Butterfly6<T> {
+    #[inline(always)]
+    pub fn new(direction: FftDirection) -> Self {
+        assert_f32::<T>();
+        let bf3 = SseF32Butterfly3::new(direction);
+
+        Self {
+            direction,
+            _phantom: std::marker::PhantomData,
+            bf3,
+        }
+    }
+
+    #[inline(always)]
+    pub(crate) unsafe fn perform_fft_contiguous(
+        &self,
+        input: RawSlice<Complex<T>>,
+        output: RawSliceMut<Complex<T>>,
+    ) {
+        let value01 = _mm_loadu_ps(input.as_ptr() as *const f32);
+        let value23 = _mm_loadu_ps(input.as_ptr().add(2) as *const f32);
+        let value45 = _mm_loadu_ps(input.as_ptr().add(4) as *const f32);
+
+        let temp = self.perform_fft_direct(value01, value23, value45);
+
+        let array = std::mem::transmute::<[__m128; 3], [Complex<f32>; 6]>(temp);
+
+        let output_slice = output.as_mut_ptr() as *mut Complex<f32>;
+
+        *output_slice.add(0) = array[0];
+        *output_slice.add(1) = array[1];
+        *output_slice.add(2) = array[2];
+        *output_slice.add(3) = array[3];
+        *output_slice.add(4) = array[4];
+        *output_slice.add(5) = array[5];
+    }
+
+    #[inline(always)]
+    pub(crate) unsafe fn perform_dual_fft_contiguous(
```

Comment:
What is a "dual FFT"? I see this phrase repeated throughout the SSE butterflies and it's not clear what it does.

Reply:
Yes it's exactly that, it's identical to the f64 version, but does two ffts in parallel.
The column butterflies also include multiplication with twiddle factors right? So then it's not quite the same. This is something simpler, just doing two ffts instead of one. I'm afraid that renaming to column something will instead make other things difficult to read, like here: https://github.com/HEnquist/RustFFT/blob/ssesimple/src/sse/sse_butterflies.rs#L33

Reply:
The ones in AVX multiply with their own internal twiddle factors, but not outer twiddle factors.

If renming these to column butterflies won't work, perhaps we should rename these and the ones in AVX to some new name? Maybe `parallel_butterflies`? Any ideas?

Reply:
Here's column_butterfly6 https://github.com/HEnquist/RustFFT/blob/ssesimple/src/avx/avx_vector.rs#L533

Reply:
"parallell" is a good word here! I think that describes very well what is going on. "dual" only works for the case of two so isn't very good, and "column" only becomes clear once you have understood the whole row-and-column thing.